src/types.rs

```diff
@@ -600,14 +609,58 @@ pub struct ContractAddress {
     serde(tag = "type", content = "address", rename_all = "lowercase")
 )]
 #[cfg_attr(feature = "fuzz", derive(Arbitrary))]
-#[derive(PartialEq, Eq, Copy, Clone, Debug)]
+#[derive(Eq, Copy, Clone, Debug)]
 pub enum Address {
     Account(AccountAddress),
     Contract(ContractAddress),
 }
 
+// This trait is implemented manually to save bytes
+impl PartialEq for Address {
+    fn eq(&self, other: &Address) -> bool {
+        match (self, other) {
```

Comment:
Is there a reason for not pattern matching on `(&self, other)` here?

Is this better or does it not make a difference.

Reply:
I am not sure what you mean by this.
Would you mind elaborating?

Reply:
Never mind this, I read it wrong.